packages/page-accounts/src/modals/CreateEthDerivationPath.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017-2021 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import React, { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+
+import { Checkbox, Dropdown, Input, InputNumber, MarkError, MarkWarning, Modal } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { DeriveValidationOutput, PairType } from '../types';
+
+interface Props {
+  className?: string;
+  onChange: (string: string) => void;
+  seedType: string;
+  derivePath: string;
+  deriveValidation: DeriveValidationOutput | undefined;
+  seed: string;
+  addressFromSeed: (seed: string, derivePath: string, pairType: PairType) => string;
+}
+
+export const ETH_DEFAULT_PATH = "m/44'/60'/0'/0/0";
+
+function CreateEthDerivationPath ({ addressFromSeed,
+  className,
+  derivePath,
+  deriveValidation,
+  onChange,
+  seed,
+  seedType }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const [addIndex, setAddIndex] = useState(0);
+  const [customIndex, setCustomIndex] = useState(new BN(0));
+  const [addressList, setAddressList] = useState<{ key: number; text: ReactNode; value: number; }[]>([{
+    key: 0,
+    text: t('Address index 0'),
+    value: 0
+  }]);
+  const [useCustomPath, setUseCustomPath] = useState(false);
+  const [useCustomIndex, setUseCustomIndex] = useState(false);
```

Comment:
There is also a `useToggle` hook from `@polkadot/react-hooks` that may be of interest here since it seems the callbacks, well, just toggle. As an example -

```js
const [isCustomPath, toggleCustomPath] = useToggle();
```

Then insteda of the callbacks defined below, just pass through `toggleCustomPath`

Reply:
thanks